lib/gems/pending/util/miq-password.rb

```diff
@@ -23,21 +23,23 @@ def encrypt(str, ver = "v2", key = self.class.keys[ver])
     "#{ver}:{#{value}}"
   end
 
-  def decrypt(str, legacy = false)
-    if str.nil? || str.empty?
-      str
-    else
-      ver, enc = self.class.split(str)
-      return "" if enc.empty?
+  def decrypt(str, legacy = false, key = nil)
+    return str if str.nil? || str.empty?
+
+    ver, enc = self.class.split(str)
+    return "" if enc.empty?
+
+    ver ||= "0" # if we don't know what it is, just assume legacy
 
-      ver ||= "0" # if we don't know what it is, just assume legacy
+    key ||= begin
       key_name = (ver == "2" && legacy) ? "alt" : "v#{ver}"
```

Comment:
ugh. who wrote that?

those are different "legacy"

line 33 means legacy aka pre 5.0 keys.

line 34 means legacy aka the v2 keys that is no longer valid (because we are actively migrating away from it) - it is stored as `:alt`